### PR TITLE
Improved error message if catalog doesnt exists

### LINKED ISSUE #178: sqlalchemy.orm.exc.NoResultFound: No row was found for one()
Problem description:

If I run detect on a Redshift database with a new catalog db specified for the run, it returns this:

piicatcher --catalog-host localhost --catalog-port 5432 --catalog-user piicatcher --catalog-password <blah> --catalog-database piicatcher_1 detect --source-name <dbname> --scan-type metadata --include-schema <schemaname>

Traceback (most recent call last):
  File "/home/ec2-user/.env/bin/piicatcher", line 8, in <module>
    sys.exit(app())
  File "/home/ec2-user/.env/lib64/python3.7/site-packages/typer/main.py", line 214, in __call__
    return get_command(self)(*args, **kwargs)
  File "/home/ec2-user/.env/lib64/python3.7/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/home/ec2-user/.env/lib64/python3.7/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/home/ec2-user/.env/lib64/python3.7/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/ec2-user/.env/lib64/python3.7/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/ec2-user/.env/lib64/python3.7/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/home/ec2-user/.env/lib64/python3.7/site-packages/typer/main.py", line 500, in wrapper
    return callback(**use_params)  # type: ignore
  File "/home/ec2-user/.env/lib64/python3.7/site-packages/piicatcher/command_line.py", line 196, in detect
    source = catalog.get_source(source_name)
  File "/home/ec2-user/.env/lib64/python3.7/site-packages/dbcat/catalog/catalog.py", line 191, in get_source
    .filter(CatSource.name == source_name)
  File "/home/ec2-user/.env/lib64/python3.7/site-packages/sqlalchemy/orm/query.py", line 3500, in one
    raise orm_exc.NoResultFound("No row was found for one()")
sqlalchemy.orm.exc.NoResultFound: No row was found for one()

Expected Result;

A new db with all the required metadata tables gets created for the scan to happen (if the table does not exist already)

### PR DESCRIPTION
**Pull request by yjagdale**
* Improved error message if catalog doesnt exists

closes #178

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ on:
 
 jobs:
   build:
-    runs-on: ubuntu-latest
```
Comment (vrajat) [inline .github/workflows/ci.yml:13]: Can you explain the reason to change the version number of ubuntu?

File: piicatcher/command_line.py [hunk 2]
```diff
@@ -88,40 +89,40 @@ def version_callback(value: bool):
...
-        None, help="database of Postgres database. Use if catalog is a database."
-    ),
-    catalog_secret: str = typer.Option(
-        dbcat.settings.DEFAULT_CATALOG_SECRET,
-        help="Secret to encrypt sensitive data like passwords in the catalog.",
-    ),
-    version: Optional[bool] = typer.Option(
-        None, "--version", callback=version_callback, is_eager=True
-    ),
+        log_level: str = typer.Option("WARNING", help="Logging Level"),
```
Comment (vrajat) [inline piicatcher/command_line.py:92]: There are whitespace changes. Can you revert these changes?

**Review by vrajat — COMMENTED** [review]
The docker file change doesn't seem correct. The docker file should build an image from the current state of the code. Otherwise a release to pypi is required before releasing a docker image.

It is also not a good idea to combine changes in the same commit. 

I suggest you revert all the changes except for the error message.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (yjagdale) [reply .github/workflows/ci.yml:13]: with the latest ubuntu test are failing with some errors
```
E                 -                'tables': [{'columns': [{'data_type': 'text',
E                 ?                                                       ^^^^
E                 +                'tables': [{'columns': [{'data_type': 'TEXT',
```

This needs to be validated, But for now to limiting ubuntu works. 

[#84-build](https://github.com/tokern/piicatcher/actions/runs/3656227930/jobs/6178423203) there with the error.

File: piicatcher/command_line.py (hunk 2 above)
Comment (yjagdale) [reply piicatcher/command_line.py:92]: Sure.

**Comment by yjagdale** [comment]
> The docker file change doesn't seem correct. The docker file should build an image from the current state of the code. Otherwise a release to pypi is required before releasing a docker image.
> 
> It is also not a good idea to combine changes in the same commit.
> 
> I suggest you revert all the changes except for the error message.

Done. Reverted all the code for docker and other changes except the error message. 

A few things need to go in like ubuntu(breaking build for that at the moment).
